src/codegen.cpp

```diff
@@ -2363,6 +2366,8 @@ std::unique_ptr<Module> jl_create_llvm_module(StringRef name, LLVMContext &conte
     if (!m->getModuleFlag("Debug Info Version"))
         m->addModuleFlag(llvm::Module::Warning, "Debug Info Version",
             llvm::DEBUG_METADATA_VERSION);
+    if (imaging_mode)
+        m->addModuleFlag(llvm::Module::Error, "julia.imaging_mode", 1);
```

Comment:
At the moment we make different choices in imaging mode vs non-imaging mode, so I wanted to carry that over (particularly in `jl_dump_native`, where we guess imaging mode from the global defaults rather than from how the code was generated).